libraries/SITL/SIM_Frame.h

```diff
@@ -44,19 +45,92 @@ class Frame {
     static Frame *find_frame(const char *name);
     
     // initialise frame
-    void init(float mass, float hover_throttle, float terminal_velocity, float terminal_rotation_rate);
-
+    void init(const char *frame_str);
+    
     // calculate rotational and linear accelerations
     void calculate_forces(const Aircraft &aircraft,
                           const struct sitl_input &input,
-                          Vector3f &rot_accel, Vector3f &body_accel, float* rpm);
+                          Vector3f &rot_accel, Vector3f &body_accel, float* rpm,
+                          bool use_drag=true);
     
     float terminal_velocity;
     float terminal_rotation_rate;
-    float thrust_scale;
     uint8_t motor_offset;
 
     // calculate current and voltage
-    void current_and_voltage(const struct sitl_input &input, float &voltage, float &current);
+    void current_and_voltage(float &voltage, float &current);
+
+    // get mass in kg
+    float get_mass(void) const {
+        return mass;
+    }
+
+private:
+    /*
+      parameters that define the multicopter model. Can be loaded from
+      a json file to give a custom model
+     */
+    const struct Model {
+        // model mass kg
+        float mass = 3.0;
+
+        // diameter of model
+        float diagonal_size = 0.35;
+
+        /*
+          the ref values are for a test at fixed angle, used to estimate drag
+         */
+        float refSpd = 15.08; // m/s
```

Comment:
normally we don't use camel case in our c++ files..

Reply:
it is used to match the json field names